computer_vision_draft/houghTransform/houghTransform/DrivabilityDetector.cpp

```diff
@@ -0,0 +1,118 @@
+#include "DrivabilityDetector.h"
+
+
+//Constructor definition
+DrivabilityDetector::DrivabilityDetector(float prevWeight, Point2i initialPoint, int averageSize) {
+	//Centerpoint
+	this->prevWeight = prevWeight;
+	this->centerPoint = initialPoint;
+
+	//Drivable rows
+	this->rowFromBottom = 0;
+	this->averageSize = averageSize;
+	this->rawRowFromBottom = vector<int>(averageSize, 0);
+
+	//Initializing images
+	this->drivabilityMap = Mat(480, 640, CV_8UC1, Scalar(0));
+	this->drivabilityMap.copyTo(this->lineImage);
+}
+
+//Function definitions
+
+
+/*
+Using Hough Lines transform to determine the drivable parts of the floor.
+* Basic process is
+* 1 - Blur image with Gaussian filter
+* 2 - Find edges using Canny edge detector
+* 3 - Find nearly horizontal lines using Hough line transform
+* 4 - Print lines onto empty array
+* 5 - Search through array until line is found, find row furthest from bottom of image
+* 6 - Calculate center row, take as centerpoint(+ weighted average with last value)
+* 7 - Pass last n values for number of rows from the bottom of the array through moving average filter
+* 8 - Return no. rows from bottom
+*/
+int DrivabilityDetector::calculateRowFromBottom(cv::Mat image) {
+	//1 - Blur image with Gaussian filter
+	Size2i kernelSize = Size(7, 7);			    //Size of kernel for convolution, larger -> smoother output + slower
+	GaussianBlur(image, image, kernelSize, 0); //Stdev calculated to fit Kernel
+
+	
+	//2 - Find edges using Canny edge detector
+	Mat edge;
+	Canny(image, edge, 25, 80);
+
+	//3 - Find nearly horizontal lines using Hough line transform
+	std::vector<Vec2f> lines;
+	double rho = 1.0;    //Pixel resolution
+	double theta = 1.0 / 180 * CV_PI; //Angle resolution
+	double minAngle = 85.0 / 180 * CV_PI;
+	double maxAngle = 105.0 / 180 * CV_PI;
+
+	int minVotes = 36; //Mininum number of votes for a line
+	HoughLines(edge, lines, rho, theta, minVotes, 0, 0, minAngle, maxAngle);
+
+	//4 - Print lines onto empty array
+	int width = image.cols;
+	int height = image.rows;
+	Mat emptyMatrix(height, width, CV_8UC1, Scalar(0)); //Initialize empty matrix, 8-bit, 1-color channel, 
+	for (size_t k = 0; k < lines.size(); k++) {
+		float rho = lines[k][0];
+		float theta = lines[k][1];
+		if (((theta > 89.5 / 180 * CV_PI) && (theta < 90.5 / 180 * CV_PI)))
+			continue;
+		double a = cos(theta), b = sin(theta);
+		double x0 = a * rho, y0 = b * rho; //Find centerpoint in x,y coordinates
+		int pixelWidth = 1000;
```

Comment:
same here, this should probably be a const

Reply:
Applied suggestion